_posts/en/newsletters/2019-01-15-newsletter.md

```diff
@@ -0,0 +1,94 @@
+popular Bitcoin infrastructure projects.
+
+## Action items
+
+- **Upgrade to C-Lightning 0.6.3:** this release fixes a remote DoS
+  vulnerability that could be used to crash C-Lightning nodes and
+  potentially steal money.  See the *notable code changes* section below
+  for details.  This release also includes other less critical bug fixes
+  and new features.
+
+## News
+
+- **Weak signature nonces discovered:** a preprint [paper][weak nonces]
+  by researchers Joachim Breitner and Nadia Heninger describes how they
+  discovered hundreds of Bitcoin private keys by looking for signatures
+  generated using nonces with less than the expected entropy of 256
+  bits.  Independent [code archaeology][gmaxwell bitcore] by Gregory
+  Maxwell indicates that the main culprit was probably the BitPay
+  Bitcore software which introduced a bug around July 2014 and released
+  a fix about a month later.  (Note: BitPay Bitcore is unrelated to
+  Bitcoin Core.)  From there, the bug propagated to software such as
+  BitPay Copay that depended upon Bitcore.  About 97% of the faulty
+  signatures found in the paper are compatible with Maxwell's Copay
+  hypothesis, and the paper provides plausible explanations for most of
+  the remaining 3% of signatures, indicating that users of modern
+  wallets are probably safe provided they do not continue to use
+  addresses whose bitcoins they spent using earlier vulnerable programs.
+
+    If you ever used an affected version of Bitcore (0.1.28 to 0.1.35),
+    Copay (0.4.1 to 0.4.3), or other vulnerable software, you should
+    create a new wallet file, send all of your funds from the old wallet
+    file to an address in the new wallet, and discontinue use of the
+    previous wallet file.  When designing software that signs Bitcoin
+    transactions, you should prefer to use peer-reviewed implementations
+    that generate signature nonces determinstically, such as
+    [libsecp256k1][] which implements [RFC6979][].
+
+## Notable code changes
+
+*Notable code changes this week in Bitcoin Core, LND, C-lightning,
+Eclair, and libsecp256k1.*
+
+- [Bitcoin Core #15039][] disables nLockTime-based anti-fee-sniping if
+  the most recent block seen by the node had a timestamp eight or more
+  hours ago.  Anti-fee-sniping attempts to equalize the advantages
+  between honest miners who simply extend the block chain and dishonest
+  miners who create chain forks in an attempt to steal fees from honest
+  miners.  However, when using anti-fee-sniping, nodes that have been
+  offline for a while don't know what block is at the tip of the chain
+  and so they could create multiple transactions offline that would all
+  use the same very old nLockTime value, linking those transactions
+  together in block chain analysis.  This merge fixes the problem by
+  disabling the feature if a node is offline for too long.
+
+- [C-Lightning #2214][] fixes a DoS vulnerability where a peer could
```

Comment:
I like all the information here, but it looks like the most important information ('upgrade is highly recommended') is buried in the middle of the text. You have highlighted the upgrade advice in the **actions** section above, but I wonder if we can also make it more visible here. Perhaps something like:

```
- [C-Lightning #2214][] fixes a remote crash bug which could lead
  to loss of funds. **All users are advised to upgrade to 0.6.3 to
  get a fix for this issue.**

  Before this fix, a peer could cause the c-lightning node to crash
  by trying to get it to accept a payment with a smaller timelock
  than the node allows. As with all lightning node remote-crash
  bugs, this could allow an attacker to steal funds from open channels.
  However, in order to do this, the attacker must risk their own money.
  This is hoped to be enough of a risk to discourage most attacks.
```